### PR TITLE
Generalizing Application.parseListeners() to be used by the Schema Registry

### PR DESCRIPTION
**Pull request by alexlod**
This change is to support https://github.com/confluentinc/schema-registry/pull/371

### CODE REVIEW

File: core/src/main/java/io/confluent/rest/Application.java [hunk 1]
```diff
@@ -260,10 +262,14 @@ protected void doStop() throws Exception {
...
-      if (scheme != null && (scheme.equals("http") || scheme.equals("https"))) {
+      if (uri.getPort() == -1) {
+        throw new ConfigException("Found a listener without a port. All listeners must have a port. The " +
+          "listener without a port is: " + listenerStr);
+      }
+      if (scheme != null && supportedSchemes.contains(scheme)) {
         listeners.add(uri);
       } else {
-        log.warn("Found a listener with an unsupported scheme (ony http and https are supported). Ignoring " +
+        log.warn("Found a listener with an unsupported scheme (supported: " + supportedSchemes + "). Ignoring " +
```
Comment (ewencp) [inline core/src/main/java/io/confluent/rest/Application.java]: Should we be throwing a ConfigException if this else clause filters out all listeners?

**Comment by ewencp** [comment]
@alexlod One minor question, but otherwise LGTM

File: core/src/main/java/io/confluent/rest/Application.java (hunk 1 above)
Comment (alexlod) [inline core/src/main/java/io/confluent/rest/Application.java]: We are -- see line 277 just below this.

Comment (ewencp) [inline core/src/main/java/io/confluent/rest/Application.java]: Ah, apparently I cannot properly match braces.

Comment (alexlod) [inline core/src/main/java/io/confluent/rest/Application.java]: :). I'm gonna merge this thang.

**Pull request merged**